src/main/java/org/jenkinsci/plugins/workflow/log/BufferedBuildListener.java

```diff
@@ -46,10 +49,23 @@ final class BufferedBuildListener extends OutputStreamTaskListener.Default imple
 
     private static final Logger LOGGER = Logger.getLogger(BufferedBuildListener.class.getName());
 
+    private static final Cleaner cleaner = Cleaner.create(new NamingThreadFactory(new DaemonThreadFactory(), BufferedBuildListener.class.getName() + ".cleaner"));
+
     private final OutputStream out;
+    private final Channel channel;
+    private final Listener listener;
```

Comment:
SpotBugs does not like these unserializable fields given that the class is `Serializable` although it does not matter in our case. I think they can be transient though:

```suggestion
    private transient final Channel channel;
    private transient final Listener listener;
```